rasa/utils/endpoints.py

```diff
@@ -149,6 +149,9 @@ async def request(
             headers.update(kwargs["headers"])
```

Comment:
Will the changes done to the headers here be "undone" by your changes?

Reply:
There is a possibility that the similar keys get overwritten here. Do you have any ideas what can be done to minimise this risk, perhaps only add the tracing related header key?

Reply:
It depends what could be in the `self.headers`, in the tracing scenarios it only contains the `traceparent` key-value pair. I've had a look at where this `EndpointConfig.request()` method is used in the codebase, and only in the `rasa interactive` code it is used in two places with `headers` keyword argument with value `{"Accept": "application/json"}` , let me know if you're interested and I can share the links.

Reply:
> only add the tracing related header key

yep sure, we can restrict to this only, however I think the risk is low for overwriting keys based on the code usage I could find of `headers` keyword argument (see reply above). Let me know before making the commit if I should go ahead?

Reply:
If you've looked into it already then it's cool. I'm good with it